src/metrics/scopus/citations.py

```diff
@@ -2,7 +2,7 @@
 import logging
 from django.conf import settings
 from metrics import models, handler
-from metrics.utils import first, flatten, simple_rate_limiter, lmap, lfilter
+from metrics.utils import first, flatten, simple_rate_limiter, lmap, lfilter, has_key, ParseError, ensure
```

Comment:
A lot of these utility functions are used in multiple projects. Do you think there are enough to move them into a small library?

Reply:
I wonder if there's a theme they can all relate to, like data structures manipulation

Reply:
I made an attempt months ago but I never pushed anything. I'm not worried so much about the duplication of functions but more the tests that must come with it and the hit to a project's coverage when I don't copy the accompanying tests.

Another thing is that these functions and their names are pretty idiomatic to _me_ and my way of programming. I know them well but fresh eyes would have to go hunt down another repository to figure out what they do.

> like data structures manipulation

that is all programming is at the end of the day. syntax just gets in the way. mostly.

Reply:
I'm mistaken, I did push them somewhere but the repo is private: https://bitbucket.org/lskibinski/lsh-utils

I'll clean it up and stick it on pypi this week. Then we can prune more code away from projects.

Reply:
heh - and it was more than a 'few months ago', more like a year and a few months ago (nov '16). time flies.

Reply:
> `lsh-utils`

I agree that the naming (functions but also repository) can be made more general in the interest of reuse from other people :)

Reply:
ha
 
> "the shell's name (lash) is an acronym for _Luke-again shell_, a pun on the name of the Luke shell (lsh) that it replaced in 1976 ..."`